src/transformers/models/data2vec/modeling_data2vec.py

```diff
@@ -0,0 +1,1577 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""PyTorch data2vec model."""
+
+import math
+
+import torch
+import torch.utils.checkpoint
+from packaging import version
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN, gelu
+from ...file_utils import (
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from ...modeling_outputs import (
+    BaseModelOutputWithPastAndCrossAttentions,
+    BaseModelOutputWithPoolingAndCrossAttentions,
+    CausalLMOutputWithCrossAttentions,
+    MaskedLMOutput,
+    MultipleChoiceModelOutput,
+    QuestionAnsweringModelOutput,
+    SequenceClassifierOutput,
+    TokenClassifierOutput,
+)
+from ...modeling_utils import (
+    PreTrainedModel,
+    apply_chunking_to_forward,
+    find_pruneable_heads_and_indices,
+    prune_linear_layer,
+)
+from ...utils import logging
+from .configuration_data2vec import Data2VecConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "data2vec"
+_CONFIG_FOR_DOC = "Data2VecConfig"
+_TOKENIZER_FOR_DOC = "RobertaTokenizer"
+
+DATA2VEC_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "data2vec",
+    # See all data2vec models at https://huggingface.co/models?filter=data2vec
+]
+
+
+# Copied from transformers.models.roberta.modeling_roberta.RobertaEmbeddings with Roberta->Data2Vec
+class Data2VecEmbeddings(nn.Module):
```

Comment:
Perfect! @patil-suraj - what do you think about the naming here? Is `Data2VecEmbeddingsForText` or `Data2VecTextEmbeddings` maybe better?

Reply:
I was thinking maybe `Data2VecForTextEmbeddings`, to match the model name `Data2VecForTextModel`. Or we can change the model name to `Data2VecTextModel`.

Reply:
works for me as well - whatever you prefer :-)